src/test/java/org/cactoos/io/TeeInputPartATest.java

```diff
@@ -0,0 +1,979 @@
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.cactoos.io;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.function.Supplier;
+import org.cactoos.text.TextOf;
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.Matchers;
+import org.junit.Test;
+
+/**
+ * Test case for {@link TeeInput}. Part A.
+ * @author Roman Proshin (roman@proshin.org)
+ * @version $Id$
+ * @since 1.0
+ * @checkstyle JavadocMethodCheck (1000 lines)
+ * @checkstyle ClassDataAbstractionCouplingCheck (1000 lines)
+ */
+@SuppressWarnings(
+    {"PMD.TooManyMethods", "PMD.AvoidDuplicateLiterals", "PMD.GodClass"}
+)
+public final class TeeInputPartATest {
+
+    /**
+     * Test content for all the tests in this class.
+     */
+    private static final String CONTENT =
+        "Hello, товарищ äÄ üÜ öÖ and ß";
+
+    @Test
+    public void copiesFromUrlToPath() throws IOException {
+        final Path input = Files.createTempFile("input", ".txt");
+        Files.write(
+            input,
+            TeeInputPartATest.CONTENT.getBytes(StandardCharsets.UTF_8)
+        );
+        final Path output = Files.createTempFile("output", ".txt");
+        this.assertThat(
+            new TeeInput(
+                input
+                    .toUri()
+                    .toURL(),
+                output
+            ),
+            TeeInputPartATest.CONTENT,
+            () -> {
+                try {
+                    return new TextOf(output).asString();
+                } catch (final IOException exception) {
+                    throw new AssertionError(
+                        "There was an IOException",
+                        exception
+                    );
+                }
+            }
+        );
+    }
+
+    @Test
+    public void copiesFromUrlToFile() throws IOException {
+        final Path input = Files.createTempFile("input", ".txt");
+        Files.write(
+            input,
+            TeeInputPartATest.CONTENT.getBytes(StandardCharsets.UTF_8)
+        );
+        final Path output = Files.createTempFile("output", ".txt");
+        this.assertThat(
+            new TeeInput(
+                input
+                    .toUri()
+                    .toURL(),
+                output.toFile()
+            ),
+            TeeInputPartATest.CONTENT,
+            () -> {
+                try {
+                    return new TextOf(output).asString();
+                } catch (final IOException exception) {
+                    throw new AssertionError(
+                        "There was an IOException",
+                        exception
+                    );
+                }
+            }
+        );
+    }
+
+    @Test
+    public void copiesFromUrlToOutput() throws IOException {
+        final Path input = Files.createTempFile("input", ".txt");
```

Comment:
@proshin-roman I think these temp files will automatically be deleted when the test is over or when the test fails/throws an exception, right?

Reply:
@amihaiemil No, these files are not deleted automatically. Method [deleteOnExit](https://docs.oracle.com/javase/7/docs/api/java/io/File.html#deleteOnExit%28%29) has to be called to enable automatic deletion on JVM shutdown (more info [here](https://stackoverflow.com/questions/16691437/when-are-java-temporary-files-deleted)). Nevertheless, I found that no tests in Cactoos care about this problem, so all temp files stay forever in OS temp directory (until it's cleared by the OS itself). I will report a bug about this problem as it has to be fixed in many tests.

Reply:
@amihaiemil I've submitted #702